IndicoV2/TokenHandler.cs

```diff
@@ -0,0 +1,82 @@
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Indico.Exception;
+
+namespace IndicoV2
+{
+    internal class TokenHandler : DelegatingHandler
```

Comment:
This duplicates some of the code here: https://github.com/IndicoDataSolutions/indico-client-csharp/blob/7080fd610b7d01bae2377c0788dc3ea7176f8c0b/IndicoV2.StrawberryShake/HttpClient/AuthenticatingMessageHandler.cs#L12C32-L12C32

I wonder if it's feasible to refactor so we're using the HttpClient we inject into StrawberryShake? That way the auth code is re-used.

Reply:
putting an AuthenticationMessageHandler in worked!